homeassistant/components/kmtronic/config_flow.py

```diff
@@ -0,0 +1,74 @@
+"""Config flow for kmtronic integration."""
+import logging
+
+import aiohttp
+from pykmtronic.auth import Auth
+from pykmtronic.hub import KMTronicHubAPI
+import voluptuous as vol
+
+from homeassistant import config_entries, core, exceptions
+from homeassistant.helpers import aiohttp_client
+
+from .const import CONF_HOSTNAME, CONF_PASSWORD, CONF_USERNAME
+from .const import DOMAIN  # pylint:disable=unused-import
+
+_LOGGER = logging.getLogger(__name__)
+
+DATA_SCHEMA = vol.Schema({CONF_HOSTNAME: str, CONF_USERNAME: str, CONF_PASSWORD: str})
+
+
+async def validate_input(hass: core.HomeAssistant, data):
+    """Validate the user input allows us to connect."""
+
+    session = aiohttp_client.async_get_clientsession(hass)
+    auth = Auth(
+        session,
+        f"http://{data[CONF_HOSTNAME]}",
+        data[CONF_USERNAME],
+        data[CONF_PASSWORD],
+    )
+    hub = KMTronicHubAPI(auth)
+
+    try:
+        await hub.async_get_status()
+    except aiohttp.client_exceptions.ClientResponseError as err:
+        raise InvalidAuth from err
+    except aiohttp.client_exceptions.ClientConnectorError as err:
+        raise CannotConnect from err
+
+    return data
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for kmtronic."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_POLL
+
+    async def async_step_user(self, user_input=None):
+        """Handle the initial step."""
+        errors = {}
+        if user_input is not None:
+            try:
+                info = await validate_input(self.hass, user_input)
+
+                return self.async_create_entry(title=info["host"], data=user_input)
```

Comment:
Please only wrap the line that can raise inside the try... except block. Use an `else:` block if needed.